### PR TITLE
completion: include bookmarks and tags in revisions

### PR DESCRIPTION
**Pull request by senekor**
# Checklist

If applicable:
- [ ] I have updated `CHANGELOG.md`
- [ ] I have updated the documentation (README.md, docs/, demos/)
- [ ] I have updated the config schema (cli/src/config-schema.json)
- [x] I have added tests to cover my changes

### CODE REVIEW

**Review by martinvonz — APPROVED** [review]
Thanks!

File: cli/src/complete.rs
Comment (yuja) [inline cli/src/complete.rs:228]: Should we omit old tags and branches? I don't think it's uncommon to query changes between some old tagged revision (specified by tag) and recent commits (specified by `@` or change id.)

Comment (senekor) [reply cli/src/complete.rs:228]: That's true. The revisions are limited to 100, any bookmarks / tags not pointing to this limited list are not shown. I'll have to shell out twice to improve that.

Comment (yuja) [reply cli/src/complete.rs:228]: Yeah, we'll need a helper struct.

BTW, for "all_revisions" query, it might make sense to use `revsets.short-prefixes` (instead of `--limit 100`) to collect change ids which the user would be interested in. For "mutable_revisions", using `mutable()` seems good.

**Comment by senekor** [comment]
Splitting the commands resulted in quite a few changes, I'll wait for another approving review before merging.

**Review by yuja — APPROVED** [review]
Thanks.

**Pull request merged**
